File: app/src/main/java/com/example/calculatorexample/MainActivity.java
package com.example.calculatorexample;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @SuppressLint("SetTextI18n")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button btnSum = (Button) findViewById(R.id.btnSummation);
        Button btnMinus = (Button) findViewById(R.id.btnMinus);
        Button btnMulti = (Button) findViewById(R.id.btnMultiplication);
        Button btnDevi = (Button) findViewById(R.id.btnDevision);

        TextView txtResult = (TextView) findViewById(R.id.txtResult);

        EditText edtFirst = (EditText) findViewById(R.id.edtFirst);
        EditText edtSecond = (EditText) findViewById(R.id.edtSecond);

        btnSum.setOnClickListener(v->{
            try {
                double a = Double.parseDouble(
                        edtFirst.getText().toString()) + Double.parseDouble(edtSecond.getText().toString()
                );
                txtResult.setText(
                        edtFirst.getText().toString() + " + " + edtSecond.getText().toString() + " = " + a
                );
            } catch (Exception e) {
                Toast.makeText(MainActivity.this, "Không để trống", Toast.LENGTH_LONG).show();
            }
        });

        btnMinus.setOnClickListener(v->{
            try {
                double a = Double.parseDouble(
                        edtFirst.getText().toString()) - Double.parseDouble(edtSecond.getText().toString()
                );
                txtResult.setText(
                        edtFirst.getText().toString() + " - " + edtSecond.getText().toString() + " = " + a
                );
            } catch (Exception e) {
                Toast.makeText(MainActivity.this, "Không để trống", Toast.LENGTH_LONG).show();
            }
        });

        btnMulti.setOnClickListener(v->{

            try {
                double a = Double.parseDouble(
                        edtFirst.getText().toString()) * Double.parseDouble(edtSecond.getText().toString()
                );
                txtResult.setText(
                        edtFirst.getText().toString() + " * " + edtSecond.getText().toString() + " = " + a
                );
            } catch (Exception e) {
                Toast.makeText(MainActivity.this, "Không để trống", Toast.LENGTH_LONG).show();
            }
        });

        btnDevi.setOnClickListener(v->{
            try {
                if(Double.parseDouble(edtSecond.getText().toString())!=0) {
                    double a = Double.parseDouble(
                            edtFirst.getText().toString()) / Double.parseDouble(edtSecond.getText().toString()
                    );
                    txtResult.setText(
                            edtFirst.getText().toString() + " / " + edtSecond.getText().toString() + " = " + a
                    );
                } else {
                    Toast.makeText(MainActivity.this, "Mẫu khác 0", Toast.LENGTH_LONG).show();
                }

            } catch (Exception e) {
                Toast.makeText(MainActivity.this, "Không để trống", Toast.LENGTH_LONG).show();
            }
        });
    }
}

File: app/src/main/java/com/htnguyen/gplxapp/MainActivity.kt
package com.htnguyen.gplxapp

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val btnSum = findViewById<View>(R.id.btnSummation) as Button
        val btnMinus = findViewById<View>(R.id.btnMinus) as Button
        val btnMulti = findViewById<View>(R.id.btnMultiplication) as Button
        val btnDevi = findViewById<View>(R.id.btnDevision) as Button
        val txtResult = findViewById<View>(R.id.txtResult) as TextView
        val edtFirst = findViewById<View>(R.id.edtFirst) as EditText
        val edtSecond = findViewById<View>(R.id.edtSecond) as EditText
        btnSum.setOnClickListener { v: View? ->
            try {
                val a =
                    edtFirst.text.toString().toDouble() + edtSecond.text.toString()
                        .toDouble()
                txtResult.text =
                    edtFirst.text.toString() + " + " + edtSecond.text.toString() + " = " + a
            } catch (e: Exception) {
                Toast.makeText(this@MainActivity, "Không để trống", Toast.LENGTH_LONG).show()
            }
        }
        btnMinus.setOnClickListener { v: View? ->
            try {
                val a =
                    edtFirst.text.toString().toDouble() - edtSecond.text.toString()
                        .toDouble()
                txtResult.text =
                    edtFirst.text.toString() + " - " + edtSecond.text.toString() + " = " + a
            } catch (e: Exception) {
                Toast.makeText(this@MainActivity, "Không để trống", Toast.LENGTH_LONG).show()
            }
        }
        btnMulti.setOnClickListener { v: View? ->
            try {
                val a =
                    edtFirst.text.toString().toDouble() * edtSecond.text.toString()
                        .toDouble()
                txtResult.text =
                    edtFirst.text.toString() + " * " + edtSecond.text.toString() + " = " + a
            } catch (e: Exception) {
                Toast.makeText(this@MainActivity, "Không để trống", Toast.LENGTH_LONG).show()
            }
        }
        btnDevi.setOnClickListener { v: View? ->
            try {
                if (edtSecond.text.toString().toDouble() != 0) {
                    val a =
                        edtFirst.text.toString().toDouble() / edtSecond.text.toString()
                            .toDouble()
                    txtResult.text =
                        edtFirst.text.toString() + " / " + edtSecond.text.toString() + " = " + a
                } else {
                    Toast.makeText(this@MainActivity, "Mẫu khác 0", Toast.LENGTH_LONG).show()
                }
            } catch (e: Exception) {
                Toast.makeText(this@MainActivity, "Không để trống", Toast.LENGTH_LONG).show()
            }
        }
    }
}